### PR TITLE
#165903538 Fix all reponses query to filter by date range

### PR DESCRIPTION
**Pull request by d-kahara**
#### What does this PR do?
- The aim of this PR is to refactor the `allRoomResponses` query to allow filtering responses by date range

#### Description of Task to be completed?
- Fix query by date range
- Refactor endpoint to reduce complexity
- Remove unnecessary duplication

#### How should this be manually tested?
- Pull this branch `bg-fix-query-all-responses-by-date-range-165903538`
- Ensure that you have a number of responses in your database
- Run the query below for filtering by date range only
```
query{
    allRoomResponses(startDate: "jan 1 2019",
     endDate: "mar 10 2019"){
        responses{
            totalResponses
            roomName
            response{
                responseId
                
            }
        }
    }
} 
```
For filtering with additional parameters run the query below
```
query{
    allRoomResponses(startDate:"may 1 2019", endDate:"may 29 2019" ,upperLimitCount:5, lowerLimitCount:1, page:1, perPage:1){
        responses{         
            totalResponses
            roomName
          	roomId
            response{
                responseId
              	createdDate
              	missingItems
              	suggestion
              	resolved
            }
        }
    pages
    hasNext
    hasPrevious
    }

```

#### Any background context you want to provide?
n/a

#### What are the relevant pivotal tracker stories?
[#165903538](https://www.pivotaltracker.com/story/show/165903538)

### Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have linted my code prior to submission
- [x] My changes generate no new warnings
- [x] Existing unit tests pass locally with my changes
- [x] Implementation works according to expectations

### Screenshots
*Filtering by date range only*
Screenshot 2019-05-22 at 15 50 13

*Filtering with additional room name argument*
Screenshot 2019-05-24 at 10 08 44

*Filtering with additional multiple parameters*
Screenshot 2019-05-29 at 11 06 23

### CODE REVIEW

File: api/response/schema_query.py [hunk 1]
```diff
@@ -154,20 +154,25 @@ def search_response_by_room(self, info, **kwargs):
...
+                responses = RoomResponse(
+                    room_id=room.id,
+                    room_name=room_name,
+                    total_responses=response_count,
+                    response=all_responses)
+                response.append(responses)
+            return response
+        else:
+            raise GraphQLError(
+                'No response for this room, enter a valid room name')
```
Comment (rajeman) [inline api/response/schema_query.py]: @d-kahara I believe this error message is what the user gets whenever there are no active rooms and the user wants to get `allRoomResponses`. I think the message does not describe the situation well enough as it implies the user entered a `room name` in the query argument. Kindly look into this

Comment (d-kahara) [reply api/response/schema_query.py]: This message is shown when the user provides a non-active `room name` as an argument in the query. See updated screenshots

**Comment by rajeman** [comment]
Look the at the screenshot below. I didn't pass any `room name` as argument yet, this is what I got. 

image

File: api/response/schema_query.py [hunk 2]
```diff
@@ -154,20 +154,25 @@ def search_response_by_room(self, info, **kwargs):
...
+                    self, room_response, room.id)
+                responses = RoomResponse(
+                    room_id=room.id,
+                    room_name=room_name,
+                    total_responses=response_count,
+                    response=all_responses)
+                response.append(responses)
+            return response
+        else:
+            raise GraphQLError(
```
Comment (Bonifase) [inline api/response/schema_query.py]: Where is the room name being passed?

Comment (d-kahara) [reply api/response/schema_query.py]: resolved

**Comment by d-kahara** [comment]
> Look at the screenshot below. I didn't pass any `room name` as argument yet, this is what I got.
> 
> image

This implementation has been changed

**Comment by d-kahara** [comment]
> Prevent the limits range from overriding the date range

Resolved

**Pull request merged**